app/policies/grove_monitor_policy.rb

```diff
@@ -0,0 +1,9 @@
+class GroveMonitorPolicy < ApplicationPolicy
+  def index?
```

Comment:
This should inherit from TeacherResourcePolicy, and then you won't need these overrides for index and show.